pages/chaoss/visualizations/contrib_importance_over_time.py

```diff
@@ -0,0 +1,478 @@
+    period_from = pd.date_range(start=start_date, end=end_date, freq=f'{step_size}m', inclusive="both")
+    # store the period_from dates in a df 
+    df_final = period_from.to_frame(index=False, name="period_from")
+    # calculate the end of each interval and store the values in a column named period_from
+    df_final['period_to'] = df_final.apply(lambda row: row.period_from +  pd.DateOffset(months = window_width), axis=1)
+
+    # dynamically calculate the contributor prolificacy over time for each of the action times and store results in df_final
+    (df_final['Commit'],
+    df_final['Issue Opened'], 
+    df_final['PR Opened'],
+    df_final['PR Comment'], 
+    df_final['PR Review']) = zip(*df_final.apply(lambda row: cntrb_prolificacy_over_time(df, row.period_from, row.period_to, window_width, threshold), axis=1))
+    
+    return df_final
+
+def create_figure(df_final, step_size):
+    # create plotly express line graph
+    fig = go.Figure(
+        [
+            go.Scatter(name = 'Commit',
+                       x = df_final['period_from'],
+                       y = df_final['Commit'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[0])
+                       ), 
+            
+            go.Scatter(name = 'Issue Opened',
+                       x = df_final['period_from'],
+                       y = df_final['Issue Opened'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[1])
+                       ), 
+
+            go.Scatter(name = 'PR Opened',
+                       x = df_final['period_from'],
+                       y = df_final['PR Opened'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[2])
+                       ),
+            
+            go.Scatter(name = 'PR Request Comment',
+                       x = df_final['period_from'],
+                       y = df_final['PR Comment'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[3])
+                       ),
+
+            go.Scatter(name = 'PR Request Review',
+                       x = df_final['period_from'],
+                       y = df_final['PR Review'],
+                       mode='lines',
+                       showlegend=True, 
+                       marker=dict(color=color_seq[4])
+                       )
+                        
+        ], 
+    )
+    
+    # define x-axis and y-axis titles and intialize first x-axis tick to start at the user-inputted start_date
+    start_date = min(df_final['period_from'])
+
+    # update xaxes to display ticks
+    fig.update_xaxes(showgrid=True,
+                     ticklabelmode="period",
```

Comment:
Still showing as slanted, personally would just do the ticks to show year or only show "Oct 20xx" every other year

Reply:
from prior comments parallel to graph = horizontal